src/UserAdmin/src/Smart.FA.Catalog.Web/Identity/UserIdentity.cs

```diff
@@ -7,26 +7,22 @@ namespace Smart.FA.Catalog.Web.Identity;
 /// <inheritdoc />
 public class UserIdentity : IUserIdentity
 {
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
     /// <inheritdoc />
-    public int Id => Identity.Id;
+    public int Id => Identity?.Id ?? 0;
 
     /// <inheritdoc />
-    public CustomIdentity Identity { get; }
+    public CustomIdentity Identity => (_httpContextAccessor.HttpContext?.User.Identity as CustomIdentity)!;
 
     /// <inheritdoc />
-    public Trainer CurrentTrainer { get; }
+    public Trainer CurrentTrainer => Identity?.Trainer!;
```

Comment:
Seeing the "Identity" method it will never return a null object
```suggestion
    public Trainer CurrentTrainer => Identity.Trainer;
```